### PR TITLE
Improve min max performance and IEEE compliance

### PR DESCRIPTION
**Pull request by CoryCharlton**
## Description
- Reduce IsNan checks to improve performance
- Fix Min not returning NaN in all cases it should
- Implement Min(float) and Max(float)

## Motivation and Context

- Fixes nanoFramework/Home#1395
- Fixes nanoFramework/Home#1396

## How Has This Been Tested?

- On device
- Unit tests

## Types of changes

- [X] Improvement (non-breaking change that improves a feature, code or algorithm)
- [X] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [X] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dev Containers (changes related with Dev Containers, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist

- [X] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [X] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [X] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).

### CODE REVIEW

**Comment by CoryCharlton** [comment]
That's all the required changes I can see so this should be stable pending any review feedback.

File: src/CLR/System.Math/nf_native_system_math_System_Math.cpp [hunk 1]
```diff
@@ -13,37 +13,55 @@ HRESULT Library_nf_native_system_math_System_Math::Max___STATIC__R8__R8__R8(CLR_
...
 #if (DP_FLOATINGPOINT == TRUE)
 
-    double x = stack.Arg0().NumericByRefConst().r8;
-    double y = stack.Arg1().NumericByRefConst().r8;
+    double val1 = stack.Arg0().NumericByRefConst().r8;
+    double val2 = stack.Arg1().NumericByRefConst().r8;
 
-    // from .NET spec: If val1, val2, or both val1 and val2 are equal to NaN, NaN is returned.
-    if (System::Double::IsNaN(x) || System::Double::IsNaN(y))
+    if (val1 != val2)
```
Comment (josesimoes) [inline src/CLR/System.Math/nf_native_system_math_System_Math.cpp:27]: We have to follow .NET here. That's what was being done. Note the comment. 
Please check the remark on the return from `Max(Double, Double)` (https://learn.microsoft.com/dotnet/api/system.math.max?view=net-7.0&WT.mc_id=DT-MVP-5004179#system-math-max(system-double-system-double)

File: src/CLR/System.Math/nf_native_system_math_System_Math.cpp [hunk 2]
```diff
@@ -13,37 +13,55 @@ HRESULT Library_nf_native_system_math_System_Math::Max___STATIC__R8__R8__R8(CLR_
...
 #else
 
-    float x = (float)stack.Arg0().NumericByRefConst().r8;
-    float y = (float)stack.Arg1().NumericByRefConst().r8;
+    float val1 = (float)stack.Arg0().NumericByRefConst().r8;
+    float val2 = (float)stack.Arg1().NumericByRefConst().r8;
 
-    // from .NET spec: If val1, val2, or both val1 and val2 are equal to NaN, NaN is returned.
-    if (System::Double::IsNaN(x) || System::Double::IsNaN(y))
+    if (val1 != val2)
```
Comment (josesimoes) [inline src/CLR/System.Math/nf_native_system_math_System_Math.cpp:50]: Same here

File: src/CLR/System.Math/nf_native_system_math_System_Math.cpp [hunk 3]
```diff
@@ -820,6 +944,16 @@ HRESULT Library_nf_native_system_math_System_Math::Max___STATIC__R8__R8__R8(CLR_
     NANOCLR_NOCLEANUP();
 }
 
+HRESULT Library_nf_native_system_math_System_Math::Max___STATIC__R4__R4__R4(CLR_RT_StackFrame &stack)
```
Comment (josesimoes) [inline src/CLR/System.Math/nf_native_system_math_System_Math.cpp:947]: This has to be removed. It's implemented at line 73.

File: src/CLR/System.Math/nf_native_system_math_System_Math.cpp [hunk 4]
```diff
@@ -830,6 +964,16 @@ HRESULT Library_nf_native_system_math_System_Math::Min___STATIC__R8__R8__R8(CLR_
     NANOCLR_NOCLEANUP();
 }
 
+HRESULT Library_nf_native_system_math_System_Math::Min___STATIC__R4__R4__R4(CLR_RT_StackFrame &stack)
```
Comment (josesimoes) [inline src/CLR/System.Math/nf_native_system_math_System_Math.cpp:967]: This has to be removed. It's implemented at line 170.

File: src/CLR/System.Math/nf_native_system_math_System_Math.cpp (hunk 1 above)
Comment (josesimoes) [reply src/CLR/System.Math/nf_native_system_math_System_Math.cpp:27]: Have you checked the Unit Test in .NET implementation about the +0 and -0? We have to be sure to follow whatever .NET has implemented, even if it's not 100% IEEE compliant.

**Pull request merged**